packages/video_player_videohole/tizen/src/drm_licence.cc

```diff
@@ -0,0 +1,767 @@
+static void HttpClose(SHttpSession* hSession);
+
+bool AppendData(SDynamicBuf* aBuf, const void* aData, size_t aSize) {
+  size_t newSize = aBuf->iSize + aSize;
+  if (aBuf->iAllocated < newSize) {
+    newSize += 1024;
+    unsigned char* buf = (unsigned char*)realloc(aBuf->iData, newSize);
+    if (!buf) {
+      LOG_ERROR("[DrmLicence] AppendData : realloc fail ");
+      return false;
+    }
+    aBuf->iData = buf;
+    aBuf->iAllocated = newSize;
+    LOG_DEBUG(
+        "[DrmLicence] AppendData : realloc aSize(%d), iSize(%d) "
+        "aBuf->iAllocated(%d)",
+        aSize, aBuf->iSize, aBuf->iAllocated);
+  }
+  memcpy(aBuf->iData + aBuf->iSize, aData, aSize);
+  aBuf->iSize += aSize;
+
+  return true;
+}
+
+char* GetRedirectLocation(const char* aHeaders, bool bSupportHttps) {
+  if (!aHeaders) {
+    return nullptr;
+  }
+
+  const char* pLocation = strcasestr(aHeaders, "Location");
+  if (!pLocation) {
+    return nullptr;
+  }
+
+  const char* ptr = pLocation + strlen("Location");
+
+  while (*ptr == ':') {
+    ptr++;
+  }
+  while (*ptr == ' ') {
+    ptr++;
+  }
+
+  unsigned i = 0;
+  while (ptr[i] && (ptr[i] != ' ') && (ptr[i] != '\n') && (ptr[i] != '\r')) {
+    i++;
+  }
+
+  if (bSupportHttps) {
+    // [soyoung] get redirection location
+    // for using https itself
+    char* ret = (char*)malloc(i + 1);
+    if (!ret) {
+      return nullptr;
+    }
+    memcpy(ret, ptr, i);
+    ret[i] = 0;
+    return ret;
+  } else {
+    // Convert Redirection Location from https to http
+    // [soyoung]
+    // Redirect location 捞 https 老 版快绰 http 肺 函券
+    // If the petition URL contains "https," the client may use SSL for the
+    // connection. (For non-SSL transport, remove the "s" in "https" from the
+    // URL.) If SSL is used, the client should check the server's certificate to
+    // ensure it is current, matches the domain, and is properly signed by a
+    // trusted authority.
+    int len = i;
+    const char* p = ptr + 4;
+    const char http_str[6] = "http\0";
+    if (i < 7) {
+      return nullptr;  // wrong location, no space even for http://
+    }
+
+    if (strncasecmp(ptr, "https", 5) == 0) {
+      len--;
+      p++;
+    }
+
+    char* ret = (char*)malloc(len + 1);
+    if (!ret) {
+      return nullptr;
+    }
+
+    memcpy(ret, http_str, 4);
+    memcpy(ret + 4, p, len - 4);
+    ret[len] = 0;
+    return ret;
+  }
+}
+
+struct curl_slist* CurlSlistAppend(struct curl_slist* aList,
+                                   const char* aString) {
+  if (!aList) {
+    return nullptr;
+  }
+
+  struct curl_slist* newList = curl_slist_append(aList, aString);
+  if (!newList) {
+    curl_slist_free_all(aList);
+  }
+
+  return newList;
+}
+
+DRM_RESULT ComposePostData_TZ(SHttpSession* hSession, const char* f_pbPostData,
+                              int f_cbPostData, const char* f_extSoapHeader) {
+  DRM_RESULT dr = DRM_SUCCESS;
+  const char* p;
+  char* dest;
+  int dest_len;
+  int remain;
```

Comment:
It is recommended to declare variables near the place where you are used.

Please name the variables so that you can know the role of variables.

Reply:
> It is recommended to declare variables near the place where you are used.
> 
> Please name the variables so that you can know the role of variables.

Fixed.